hybrid/compartment_examples/inter_comp_call_secure/main.c

```diff
@@ -146,22 +151,25 @@ void executive_switch(struct comp c)
 
 	switch_compartment();
 }
-
-void add_comp(uint8_t *_start_addr, void (*_comp_fn)())
+void add_comp(uint8_t *_start_addr, void (*_comp_fn)(), void * _comp_fn_end)
 {
-	assert(id < COMP_COUNT);
-
-	struct comp new_comp;
-	new_comp.id = id;
+    assert(id < COMP_COUNT);
+    struct comp new_comp; new_comp.id = id;
 
-	new_comp.stack_addr = (void *) (_start_addr + total_comp_size - comp_stack_size);
+	new_comp.compartment_start = (void *) _start_addr;
+	new_comp.stack_addr = (void *) (_start_addr + comp_stack_size);
 	new_comp.stack_len = comp_stack_size;
-	new_comp.heap_addr = (void *) _start_addr;
 	new_comp.heap_len = total_comp_size - comp_stack_size;
 
+    // Ensure 16-byte alignment throught the compartment bounds
+    assert(((uintptr_t) new_comp.compartment_start) % 16 == 0);
+    assert(((uintptr_t) new_comp.stack_addr) % 16 == 0);
+    assert(total_comp_size % 16 == 0);
+
     // When creating a compartment, store a local copy of the capability which
     // will allow us to call `switch_compartment` in the heap of the compartment.
-    memcpy(new_comp.heap_addr, &switcher_call, sizeof(void *__capability));
+    void * heap_top = (void *) (_start_addr + total_comp_size - sizeof(void* __capability));
```

Comment:
Minor formatting: s/`void * heap_top`/`void *heap_top`/ and s/`void* __capability`/`void *__capability`/.

Reply:
These should be fixed once I apply `clang-format`, which I have not quite yet. Should be shortly.

Reply:
Applied `clang-format` in https://github.com/capablevms/cheri-examples/pull/55/commits/5d529edf16f81593d7d5c93db5ab868e4937f978.